File: RadioJavan/Classes/Helpers/UriCreator.cs
using RadioJavan.API;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RadioJavan.Classes.Helpers
{
    internal class UriCreator
    {
        private static readonly Uri BaseRadioJavanUri = new Uri(RadioJavanApiConstants.RADIOJAVAN_URL);

        public static Uri GetLoginUri()
        {
            if (!Uri.TryCreate(BaseRadioJavanUri, RadioJavanApiConstants.LOGIN, out var uri))
                throw new Exception("Cant create URI for user login");
            return uri;
        }

        public static Uri GetForgotPasswordUri()
        {
            if (!Uri.TryCreate(BaseRadioJavanUri, RadioJavanApiConstants.FORGOT, out var uri))
                throw new Exception("Cant create URI for Forgot Password");
            return uri;
        }
    }
}

File: Dart/lib/src/Classes/Helper/UriCreator.dart
import 'package:RadioJavanDartAPI/src/API/RadioJavanApiConstants.dart';

class UriCreator {
  static final Uri BaseRadioJavanUri =
      Uri.parse(RadioJavanApiConstants.RADIOJAVAN_URL);

  static Uri GetLoginUri() => Uri.parse(
      RadioJavanApiConstants.RADIOJAVAN_URL + RadioJavanApiConstants.LOGIN);

  static Uri GetForgotPasswordUri() => Uri.parse(
      RadioJavanApiConstants.RADIOJAVAN_URL + RadioJavanApiConstants.FORGOT);

  static Uri GetRegisterUri() => Uri.parse(
      RadioJavanApiConstants.RADIOJAVAN_URL + RadioJavanApiConstants.REGISTER);
}
